packages/cli/src/ErrorHandling.ts

```diff
@@ -0,0 +1,41 @@
+import * as Sentry from '@sentry/node';
+import { RewriteFrames } from '@sentry/integrations';
+import type { Application } from 'express';
+import config from '../config';
+
+let initialized = false;
+
+export const initErrorHandling = (app?: Application) => {
+	if (initialized) return;
+
+	if (!config.getEnv('diagnostics.enabled')) {
+		initialized = true;
+		return;
+	}
+
+	const dsn = config.getEnv('diagnostics.config.sentry.dsn');
+	const { N8N_VERSION: release, ENVIRONMENT: environment } = process.env;
+
+	Sentry.init({
+		dsn,
+		release,
+		environment,
+		integrations: (defaults) => {
+			const integrations = defaults.filter(
+				(i) => !['Console', 'Modules', 'Context', 'LinkedErrors'].includes(i.name),
+			);
+			integrations.push(new RewriteFrames({ root: process.cwd() }));
+			return integrations;
+		},
+	});
+
+	if (app) {
+		const { requestHandler, errorHandler } = Sentry.Handlers;
+		app.use(requestHandler());
+		app.use(errorHandler());
+	}
+
+	initialized = true;
+};
+
+export const captureError = Sentry.captureException;
```

Comment:
> Note that these functions will not perform any action before you have called init()


Since the captureException will not do anything without calling the init function first, should no we error if the captureError method when initialized is false?

```
export const captureError = (error: Error) => {
if (!initialized) throw new Error('Call initErrorHandling before using captureError')
Sentry.captureException(error)
}
```

Reply:
we will be currently losing some errors because of this, like errors from DB init.
But, I'm doing a refactor on the the `commands` folder that will ensure that we init Sentry SDK as one of the first things. I'll create a separate PR for that next week.